syntax = "proto3";
package MyMath;

option cc_generic_services=true;

message ResultCode
{
    int32 errcode=1;
    bytes errmsg=2;
}

message AddRequest
{
    int32 num1=1;
    int32 num2=2;
}


message AddResponse
{
    ResultCode result=1;
    int32 ret=2;
}

message SubRequest
{
    int32 num1=1;
    int32 num2=2;
}


message SubResponse
{
    ResultCode result=1;
    int32 ret=2;
}

service MyMathService
{
    rpc Add(AddRequest)returns(AddResponse);
    rpc Sub(SubRequest)returns(SubResponse);
}
// 使用  protoc xxx.proto --cpp_out=./  生成.h和.cc文件